test/unit/three_scale/swagger/specification_test.rb

```diff
@@ -26,12 +36,39 @@ class ThreeScale::Swagger::SpecificationTest < ActiveSupport::TestCase
     assert specification.valid?, 'specification should be valid'
   end
 
-  test 'oas 3.0 base path' do
-    specification = ThreeScale::Swagger::Specification.new({ openapi: '3.0.0', servers: [{ url: 'https://my-petstore.io' }] }.to_json)
-    assert_equal 'https://my-petstore.io', specification.base_path
+  test 'petstore oas 3.1' do
+    specification = fixture_spec('pet-store-3.1')
+    assert specification.valid?, 'specification should be valid'
```

Comment:
Another nitpick :)

We have `assert_valid` that might work nicely here. Not sure whether error reporting will be compatible though, has to be tested. But also we have the nice `assert_predicate specification, :valid?` that will also show a nice error message, although not as complete as the `assert_valid` would (in case it works).